uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -678,14 +685,62 @@ define([
         toAPIData = leaveRequest.toAPI();
       });
 
-      it('filters out the `balance_change` and `dates` properties', function () {
+      it('filters out custom properties on leave request instance', function () {
         expect(Object.keys(toAPIData)).toEqual(_.without(
           Object.keys(leaveRequest.attributes()),
           'balance_change',
           'dates',
-          'comments'
+          'comments',
+          'uploader'
         ));
       });
     });
+
+    describe('uploading files', function () {
+      var promise;
+
+      beforeEach(function () {
+        requestData = helper.createRandomLeaveRequest();
+        instance = LeaveRequestInstance.init(requestData, false);
+        spyOn(instance.uploader, 'uploadAll').and.callThrough();
+      });
+
+      describe('on create()', function () {
+        beforeEach(function () {
+          promise = instance.create();
+        });
+
+        afterEach(function () {
+          $rootScope.$apply();
+        });
+
+        it('calls corresponding end point', function () {
+          promise.then(function () {
+            expect(instance.uploader.uploadAll).toHaveBeenCalledWith({ entityID: jasmine.any(String) });
+          });
+        });
+      });
+
+      describe('on update()', function () {
+        beforeEach(function () {
+          requestData = helper.createRandomLeaveRequest();
+          requestData.id = '12';
+          instance = LeaveRequestInstance.init(requestData, false);
+          spyOn(instance.uploader, 'uploadAll').and.callThrough();
```

Comment:
the instance is not shared across describe so the one in oncreate is not available in onupdate, hence need to create it again with id